blocks/library/heading/index.js

```diff
@@ -90,6 +108,40 @@ registerBlockType( 'core/heading', {
 					} );
 				},
 			},
+			...'23456'.split( '' ).map( ( level ) => ( {
+				type: 'shortcut',
+				shortcut: level,
+				transform( attributes ) {
```

Comment:
Sure. I was following what had been done before for multi selection transform on e.g. lists, where an array of attributes was passed as opposed to an array of blocks. See #3357 (cc @jorgefilipecosta @youknowriad). I'm fine with passing blocks instead of an array of attribute sets.